src/detect-app-layer-protocol.c

```diff
@@ -86,6 +86,7 @@ static int DetectAppLayerProtocolPacketMatch(
             p->pcap_cnt,
             p->flowflags & (FLOW_PKT_TOCLIENT|FLOW_PKT_TOSERVER),
             f->alproto, f->alproto_ts, f->alproto_tc);
+        SCReturnInt(0);
```

Comment:
Ok here, but something is missing in `PrefilterPacketAppProtoMatch` ?

Reply:
Good catch. I assumed but didn't check that prefilter wasn't supported in 7

Reply:
Looks like the function is there :-p

Reply:
Yeah, also, it looks like it has more issues, like not using `AppProtoEquals`

Reply:
addressed both in #11751